[IMP] Inventory: Batch Picking 18 (#11748)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping_receiving/picking_methods/batch.rst

demand. This method increases efficiency by allowing workers to pick items for multiple orders in
one trip through the warehouse, reducing travel time and boosting productivity.

Configuration
=============

To activate the batch picking option, begin by going to :menuselection:`Inventory app -->
Configuration --> Settings`. Under the :guilabel:`Operations` section, check the :guilabel:`Batch,
Wave & Cluster Transfers` box.

.. image:: batch/batch-transfer-checkbox.png
   :alt: Enable the *Batch Transfers* in Inventory > Configuration > Settings.

Since batch picking is a method to optimize the *pick* operation in Odoo, the :guilabel:`Storage
Locations` and :guilabel:`Multi-Step Routes` options under the :guilabel:`Warehouse` heading must
also be checked on this settings page. When finished, click :guilabel:`Save`.

.. image:: batch/locations-routes-checkbox.png
   :alt: Enable *Storage Locations* and *Multi-Step Routes* Inventory > Configuration > Settings.

.. seealso::
   - :doc:`Delivery in two steps <../daily_operations/receipts_delivery_two_steps>`
   - :doc:`../daily_operations/delivery_three_steps`

Create batch transfers
======================

> Manually create batch transfers directly from the :menuselection:`Inventory app`. On the
> :guilabel:`Receipts` Kanban card, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)`
> icon, then select :guilabel:`Prepare Batch`.

Reviewer: The way it's written, it's implying that we should only create batches for receipts. How about something more suggestive, like:
```suggestion
To manually group transfers directly from the :menuselection:`Inventory app`, hover over the desired operation type from the Inventory Overview menu (e.g. the
:guilabel:`Receipts` Kanban card), click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)`
icon, then select :guilabel:`Prepare Batch`.
```

.. image:: batch/prepare-batch.png

>    :alt: The Inventory dashboard with the Prepare Batch option highlighted.

Reviewer: After defining the Responsible, Operation Type, and Scheduled Date fields, can possibly add a note saying that the reader can learn more about the Dock Location, Vehicle, and Vehicle Category fields in the Dispatch management system documentation: https://www.odoo.com/documentation/18.0/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/dispatch.html

one trip through the warehouse, reducing travel time and boosting productivity.

Configuration
=============

To activate the batch picking option, begin by going to :menuselection:`Inventory app -->
Configuration --> Settings`. Under the :guilabel:`Operations` section, check the :guilabel:`Batch,
Wave & Cluster Transfers` box.

.. image:: batch/batch-transfer-checkbox.png
   :alt: Enable the *Batch Transfers* in Inventory > Configuration > Settings.

Since batch picking is a method to optimize the *pick* operation in Odoo, the :guilabel:`Storage
Locations` and :guilabel:`Multi-Step Routes` options under the :guilabel:`Warehouse` heading must
also be checked on this settings page. When finished, click :guilabel:`Save`.

.. image:: batch/locations-routes-checkbox.png
   :alt: Enable *Storage Locations* and *Multi-Step Routes* Inventory > Configuration > Settings.

.. seealso::
   - :doc:`Delivery in two steps <../daily_operations/receipts_delivery_two_steps>`
   - :doc:`../daily_operations/delivery_three_steps`

Create batch transfers
======================

To manually group transfers directly from the :menuselection:`Inventory app`, hover over the

> desired operation type from the Inventory Overview menu (e.g. the :guilabel:`Receipts` Kanban card),
> click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)` icon, then select :guilabel:`Prepare
> Batch`.

Reviewer 2: Please add a guilabel to this page's title :)
```suggestion
desired operation type from the :guilabel:`Inventory Overview` menu (e.g. the :guilabel:`Receipts`
Kanban card), click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)` icon, then select
:guilabel:`Prepare Batch`.
```

   :alt: Enable the *Batch Transfers* in Inventory > Configuration > Settings.

Since batch picking is a method to optimize the *pick* operation in Odoo, the :guilabel:`Storage
Locations` and :guilabel:`Multi-Step Routes` options under the :guilabel:`Warehouse` heading must
also be checked on this settings page. When finished, click :guilabel:`Save`.

.. image:: batch/locations-routes-checkbox.png
   :alt: Enable *Storage Locations* and *Multi-Step Routes* Inventory > Configuration > Settings.

Lastly, enable the warehouse picking feature, by navigating to the warehouse settings page, which is
accessible from :menuselection:`Inventory app --> Configuration --> Warehouses`.

.. seealso::
   - :doc:`Delivery in two steps <../daily_operations/receipts_delivery_two_steps>`
   - :doc:`../daily_operations/delivery_three_steps`

Create batch transfers
======================

Manually create batch transfers directly from the :menuselection:`Inventory app`. On the
:guilabel:`Receipts` Kanban card, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)`
icon, then select :guilabel:`Prepare Batch`. Click the :guilabel:`New` button to begin creating a

> batch transfer.

Reviewer 3: After I clicked 'Prepare Batch', a new batch transfer form automatically populated. There is still a 'New' button, but when I click it, it just refreshes to show the same form (screenshot).
![image](https://github.com/user-attachments/assets/d17786d1-5813-443d-abcc-98f9fb27b1ac)

.. image:: batch/prepare-batch.png
   :alt: The Inventory dashboard with the Prepare Batch option highlighted.

> On the batch transfer form, fill the following fields out accordingly:

Reviewer 3: Might be good to leave a note that some of these fields need to be made available by ticking the :icon:`oi-settings-adjust` :guilabel:`(adjust)` icon

>    :guilabel:`Operation Type`. The :guilabel:`Scheduled Date` is set to `August 11`.

Reviewer 3: In the example admonition, I would make it a full sentence by adding a couple words at the beginning, e.g. "Here is a new batch transfer..."

of the :guilabel:`Transfers` to open a filtered list of transfers.

.. image:: batch/transfers-drop-down.png

>    :align: center

Reviewer: This image (that I added when I wrote this, so this is more a critique of myself) is confusing, because the instructions say to select any of the transfers (e.g., Receipts, Deliveries, Internal, Dropships), but the highlighted selection in the image is "Batch Transfers". How about replacing this image, with something highlighting one of the options in the Transfers section?

![image](https://github.com/user-attachments/assets/4bd6ad23-b892-45a9-83bc-770d4b196bb6)

Choose from the two radio options to add to :guilabel:`an existing batch transfer` or create
:guilabel:`a new batch transfer`.

Add a :guilabel:`Description` for this batch.

.. tip::
   The :guilabel:`Description` field can be used to add additional information to help workers
   identify the source of the batch, where to place the batch, what shipping containers to use, etc.

> 

Reviewer: In line 120, we can add a little more context as to why a user would select "Draft". How about:

To create a batch to be processed later, select the Draft checkbox.

Conclude the process by clicking :guilabel:`Confirm`.

.. image:: batch/add-to-batch-window.png
   :alt: Show *Add to batch* window to create a batch transfer.

Automatic batches
-----------------

Batches can be automatically created and assigned based on several criteria. The *Automatic Batches*
option is defined on the *operation type* level.

.. example::
   In a multi-steps delivery process, the picking operation can be grouped by customer, while the
   shipping operation can be organized by carrier and destination country.

To enable *Automatic Batches*, navigate to :menuselection:`Inventory app --> Configuration -->
Operation Types`, and select the desired operation type (e.g. Delivery, Pick, etc). Then, select one
or more :guilabel:`Batch Grouping` criteria by ticking the appropriate checkbox. Even if more than
one grouping option is selected, only one batch is created.

Batches can be automatically generated based on the following criteria:

- :guilabel:`Contact`
- :guilabel:`Carrier`
- :guilabel:`Destination Country`
- :guilabel:`Source Location`
- :guilabel:`Destination Location`

.. image:: batch/auto-batch-grouping.png

>    :alt: Add alt text.

Reviewer: Remember to add alt text!

Choose from the two radio options to add to :guilabel:`an existing batch transfer` or create
:guilabel:`a new batch transfer`.

Add a :guilabel:`Description` for this batch.

.. tip::
   The :guilabel:`Description` field can be used to add additional information to help workers
   identify the source of the batch, where to place the batch, what shipping containers to use, etc.

To begin with a draft, select the :guilabel:`Draft` checkbox.

Conclude the process by clicking :guilabel:`Confirm`.

.. image:: batch/add-to-batch-window.png
   :alt: Show *Add to batch* window to create a batch transfer.

Automatic batches
-----------------

Batches can be automatically created and assigned based on several criteria. The *Automatic Batches*
option is defined on the *operation type* level, which enables the creation of batches with distinct

> grouping criteria for each operation type.

Reviewer 3: I think the readability of this explanation could be made more clear. Not sure how I would do it exactly, but I think what's sticking out to me is the circular logic of "...operation type level, which enables creation...for each operation type (level)"

Choose from the two radio options to add to :guilabel:`an existing batch transfer` or create
:guilabel:`a new batch transfer`.

Add a :guilabel:`Description` for this batch.

.. tip::
   The :guilabel:`Description` field can be used to add additional information to help workers
   identify the source of the batch, where to place the batch, what shipping containers to use, etc.

To create a batch to be processed at a later time, select the :guilabel:`Draft` checkbox.

Conclude the process by clicking :guilabel:`Confirm`.

.. image:: batch/add-to-batch-window.png
   :alt: Show *Add to batch* window to create a batch transfer.

Automatic batches
-----------------

Batches can be automatically created and assigned based on several criteria. The *Automatic Batches*
option is defined on the *operation type* level.

.. example::
   In a multi-steps delivery process, the picking operation can be grouped by customer, while the
   shipping operation can be organized by carrier and destination country.

To enable *Automatic Batches*, navigate to :menuselection:`Inventory app --> Configuration -->

> Operation Types`, and select the desired operation type (e.g. Delivery, Pick, etc). Then, select one
> or more :guilabel:`Batch Grouping` criteria by ticking the appropriate checkbox. Even if more than
> one grouping option is selected, only one batch is created.

Reviewer 2: I _believe_ these should be guilabels if they are records in list view 
```suggestion
Operation Types`, and select the desired operation type (e.g. :guilabel:`Delivery`,
:guilabel:`Pick`, etc). Then, select one or more :guilabel:`Batch Grouping` criteria by ticking the
appropriate checkbox. Even if more than one grouping option is selected, only one batch is created.
```

.. example::
   The product, `Desk Pad`, from the same batch as the :ref:`example above
   <inventory/management/batch-transfers-example>`, is only visible in the :guilabel:`Operations`

>    tab because there are no :guilabel:`Reserved` quantities in-stock to fulfill the batch transfer.

Reviewer 3: ```suggestion
   tab because there are no :guilabel:`Reserved` quantities in stock to fulfill the batch transfer.
```

Reviewer 3: Removing hyphen since it's not a modifier of a noun in this case.

This pop-up window provides the option: :guilabel:`Create Backorder?`.

Clicking the :guilabel:`Create Backorder` button automatically creates a new batch transfer.

.. note::
   When creating a new backorder, the transfers that have **not** been validated in the batch will
   be removed from it.

Click :guilabel:`No Backorder` to finish the picking *without* creating another batch picking.

Click :guilabel:`Discard` to cancel the validation, and return to the batch transfer form.

.. image:: batch/create-backorder.png
   :alt: Show the *Create Backorder* pop-up.

.. _inventory/warehouses_storage/barcode_picking:

Process batch transfer: Barcode app

> -----------------------------------

Reviewer: Similar to my feedback with the [cluster picking PR](https://github.com/odoo/documentation/pull/11753/files#r1926036497), the barcode portion currently reflects version 17, not 18. 

Can you update this section? (:

Author: Going to remove the barcode section to avoid repeating too much work. I think a single Barcode doc covering this topic for all of the picking types makes more sense
